app/controllers/users_controller.rb

```diff
@@ -11,22 +10,68 @@ def index
     @users = User.all
   end
 
+  def new
+    if current_user.admin?
+      @user = User.new
+    else
+      set_flash_message :alert, :must_be_admin
+      redirect_to current_user
+    end
+  end
+
   def create
     @user = User.new(user_params)
-
     if @user.save
-      flash[:success] = "\"#{@user.name}\" added -- Welcome to the Motley Tones"
-      redirect_to user_path(@user.id)
+      total = current_user.class.all.count
+      flash[:notice] = "New pirate added.  There #{(total == 1) ? "is" : "are"} now #{total} on the roster"
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+
+  def edit
+    if current_user.id.to_s == params[:id] || current_user.admin?
+      @user = User.find(params[:id])
     else
-      render :new
+      set_flash_message :alert, :must_be_admin
+      redirect_to current_user
     end
   end
 
-  private
+  def update
+    @user = User.find(params[:id])
+
+    if params[:user][:password].blank?
+      params[:user].delete(:password)
+      params[:user].delete(:password_confirmation)
+    end
+    if @user.update(user_params)
+      redirect_to users_path
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    if current_user.id.to_s == params[:id]
+      set_flash_message :alert, :cannot_delete_own_account
```

Comment:
once again, this could be an exception. It could also be part of an ability-based implementation. Like Cancan(can), but you can hand-roll the critical parts without bothering with the whole gem.